integration/common/openlineage/common/provider/dbt/structured_logs.py

```diff
@@ -0,0 +1,508 @@
+# Copyright 2018-2024 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+import json
+import os
+import subprocess
+from collections import defaultdict
+from functools import cached_property
+from typing import Dict, Generator, List, Optional
+
+from openlineage.client.event_v2 import RunEvent, RunState
+from openlineage.client.facet_v2 import error_message_run, job_type_job, sql_job
+from openlineage.client.uuid import generate_new_uuid
+from openlineage.common.provider.dbt.local import DbtLocalArtifactProcessor
+from openlineage.common.provider.dbt.processor import (
+    DbtVersionRunFacet,
+    ModelNode,
+    ParentRunMetadata,
+    UnsupportedDbtCommand,
+)
+from openlineage.common.provider.dbt.utils import (
+    HANDLED_COMMANDS,
+    generate_run_event,
+    get_dbt_command,
+    get_dbt_profiles_dir,
+    get_event_timestamp,
+    get_job_type,
+    get_node_unique_id,
+    get_parent_run_metadata,
+)
+from openlineage.common.utils import (
+    add_command_line_arg,
+    add_or_replace_command_line_option,
+    get_from_nullable_chain,
+)
+
+
+class DbtStructuredLogsProcessor(DbtLocalArtifactProcessor):
+    should_raise_on_unsupported_command = True
+
+    def __init__(
+        self,
+        dbt_command_line: List[str],
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.dbt_command_line: List[str] = dbt_command_line
+        self.profiles_dir: str = get_dbt_profiles_dir(command=self.dbt_command_line)
+
+        self.node_id_to_ol_run_id: Dict[str, str] = defaultdict(lambda: str(generate_new_uuid()))
+
+        # sql query ids are incremented sequentially per node_id
+        self.node_id_to_sql_query_id: Dict[str, int] = defaultdict(lambda: 1)
+        self.node_id_to_sql_start_event: Dict[str, RunEvent] = {}
+
+        self.node_id_to_inputs: Dict[str, List[ModelNode]] = {}
+        self.node_id_to_output: Dict[str, ModelNode] = {}
+
+        # will be populated when some dbt events are collected
+        self._compiled_manifest: Dict = {}
+        self._dbt_version: Optional[str] = None
+
+    @cached_property
+    def dbt_command(self) -> str:
+        return get_dbt_command(self.dbt_command_line)
+
+    @property
+    def dbt_version(self) -> Optional[str]:
+        """
+        Extracted from the first structured log MainReportVersion
+        """
+        return self._dbt_version
+
+    @cached_property
+    def catalog(self) -> Optional[Dict]:
+        if os.path.isfile(self.catalog_path):
+            return self.load_metadata(self.catalog_path, [1], self.logger)
+        return None
+
+    @cached_property
+    def profile(self):
+        profile_dict = self.load_yaml_with_jinja(os.path.join(self.profiles_dir, "profiles.yml"))[
```

Comment:
```suggestion
        profile_dict = self.load_yaml_with_jinja(os.path.expanduser(os.path.join(self.profiles_dir, "profiles.yml")))[
```

`self.profiles_dir` may contain `~` and it needs to be expanded so that it works.